js/execution.js

```diff
@@ -533,13 +687,18 @@ window.WaterbearProcess = (function () {
         this.lastTime = currTime;
 
         if (!this.paused && this.delay === 0) {
+            /* Running normally. */
             /* Run ALL of the frame handlers synchronously. */
             this.perFrameHandlers.forEach(function (strand){
                 strand.startSync();
             });
-        } else {
-            /* TODO: Run one step when paused or in slow down mode. */
-            throw new Error('Slow down frame handler not implemented.');
+        } else if (this.shouldStep && this.currentStrand === null) {
+            /* Running is delayed or paused. Run one step. */
+            this.shouldStep = false;
+            frameStrand.doNext();
+            /* XXX: ugh, this is ugly... */
```

Comment:
Let's keep flags to TODO and FIXME. I wouldn't know to search for XXX (and I wouldn't expect it to lead to a hacky solution).